packages/gatsby-cli/src/reporter/loggers/ink/cli.js

```diff
@@ -51,6 +53,26 @@ class CLI extends React.Component {
       )
     }
 
+    /*
+      Only operation on messages array is to push new message into it. Once
+      message is there it can't change. Because of that we can do single
+      transform from message object to react element and store it.
+      This will avoid calling React.createElement completely for every message
+      that can't change.
+    */
+    if (messages.length > this.messages.length) {
```

Comment:
I'm so confused about `messages` vs `this.messages`. Could we maybe name one of them differently?

Reply:
For sure! any suggestions? `this.memoizedMessageReactElements`? ;)

Reply:
I made it `this.memoizedReactElementsForMessages` - at least for now (still can have better name, but should be at least easier to review in meantime :P)

Reply:
Yeah anything is fine, just not the same :)